src/table/thead.tsx

```diff
@@ -43,6 +43,7 @@ export interface TheadProps {
   onFocusedComponentChange?: (focusId: null | string) => void;
   tableRole: TableRole;
   isExpandable?: boolean;
+  setLastUserAction?: (name: string) => void;
```

Comment:
1. It does not seem ever undefined, why is `?:` needed
2. Even if you want to make it possibly undefined, let's use `((name: string) => void) | undefined` for reasons explained here: https://github.com/cloudscape-design/components/pull/1697#discussion_r1376059808
3. If I were working on this task, I would probably extract `UserInteractionContext` and `useUserInteraction` hook to allow pulling this helper where needed inside the table

Reply:
That's unfortunate misalignment between components. I guess it stays as a reminder for future changes